### PR TITLE
fix(lodash/intersection): replace with custom implementation

### PR DESCRIPTION
**Pull request by tkrugg**
I reverted the commit replacing intersection than copie the current implem into a new `intersection` function
I add something to avoid duplicates and some tests.

### CODE REVIEW

File: src/SearchParameters/index.js [hunk 1]
```diff
@@ -1265,16 +1264,12 @@ SearchParameters.prototype = {
    * @return {string[]}
    */
   getRefinedHierarchicalFacets: function getRefinedHierarchicalFacets() {
-    var hierarchicalRefinedFacets = Object.keys(
-      this.hierarchicalFacetsRefinements
+    return intersection(
+      // enforce the order between the two arrays,
```
Comment (samouss) [inline src/SearchParameters/index.js]: I'm not completely sure to understand the comment. What does it mean?

**Comment by Haroenv** [comment]
I'm sure this has a reason, but what was wrong with the version without helper function?

**Comment by samouss** [comment]
@Haroenv The previous version didn't remove the duplicates, but the Lodash version does.

File: src/SearchParameters/index.js (hunk 1 above)
Comment (Haroenv) [reply src/SearchParameters/index.js]: This comment was there originally by the way

Comment (samouss) [reply src/SearchParameters/index.js]: Ah. It doesn't make sense to me anymore.

**Comment by Haroenv** [comment]
maybe best to also add a test to its usage to show what this is fixing

File: src/SearchParameters/index.js (hunk 1 above)
Comment (Haroenv) [reply src/SearchParameters/index.js]: I think we can remove this comment here

Comment (tkrugg) [reply src/SearchParameters/index.js]: The output is an array that contains elements from both passed arrays, so we might wonder which of the two arrays dictates the order.

In the case of [lodash intersection function](https://lodash.com/docs/4.17.11#intersection), it's the first array. 
Example:
```js
intersection([1, 2, 3, 4, 5], [5, 3]) === [3, 5] 
intersection([1, 2, 3, 4, 5], [3, 5]) === [3, 5]
intersection([1, 2, 5, 4, 3], [3, 5]) === [5, 3]
```


Whoever used intersection here wanted to keep the order of `this.hierarchicalFacets.map(...)` over the the order of `Object.keys(this.hierarchicalFacetsRefinements)`.

**Pull request merged**